torchtext/utils.py

```diff
@@ -178,7 +178,7 @@ def extract_archive(from_path, to_path=None, overwrite=False):
             for file_ in tar:
                 file_path = os.path.join(to_path, file_.name)
                 if file_.isfile():
-                    files.append(file_path)
+                    files.append(file_.name)
```

Comment:
Will this change introduce any BC breaking?

Reply:
Yes it will, but currently this function has different outputs for zip and tar files. WMT dataset is tar.gz given how _setup_datasets work, it appends the root to the extracted files, this will add the root twices for tar extracted files. 

I think it’s best to correct the function rather than include an if/else for non zip files

Reply:
Resolved by #692